### PR TITLE
dev-libs/libsecp256k1: Bump to 0.1_pre20190401

### PR DESCRIPTION
**Pull request by luke-jr**
It was released with Bitcoin Core 0.19

Closes: https://bugs.gentoo.org/720118

pkgcheck says:

```
  RequiredUseDefaults: version 0.1_pre20190401: profile: 'default/linux/arm/17.0' (32 total) failed REQUIRED_USE: experimental
```

Not really sure what to do about that...

### CODE REVIEW

File: dev-libs/libsecp256k1/libsecp256k1-0.1_pre20190401.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools eutils
```
Comment (juippis) [inline dev-libs/libsecp256k1/libsecp256k1-0.1_pre20190401.ebuild:6]: Doesn't look like *eutils.eclass* is being used here.

File: dev-libs/libsecp256k1/libsecp256k1-0.1_pre20190401.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		$(use_enable endomorphism)  \
+		--with-asm=$asm_opt \
+		--with-bignum=$(usex gmp gmp no) \
+		$(use_enable recovery module-recovery) \
+		--disable-static
+}
+
+src_install() {
+	dodoc README.md
+	emake DESTDIR="${D}" install
```
Comment (juippis) [inline dev-libs/libsecp256k1/libsecp256k1-0.1_pre20190401.ebuild:69]: These lines can be replaced with `default`.

File: dev-libs/libsecp256k1/libsecp256k1-0.1_pre20190401.ebuild [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	asm? ( || ( amd64 arm ) arm? ( experimental ) )
+	ecdh? ( experimental )
+	java? ( ecdh )
+	test-openssl? ( test )
+"
+RDEPEND="
+	gmp? ( dev-libs/gmp:0= )
+"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
```
Comment (juippis) [inline dev-libs/libsecp256k1/libsecp256k1-0.1_pre20190401.ebuild:30]: This is a BDEPEND with EAPI-7. I have a feeling the `virtual/jdk` may also be.

**Comment by luke-jr** [comment]
I'd rather not enable `experimental` by default for any platform... it's experimental for a reason. :)

Is it okay to simply insist the user decide to either set it explicitly or unset `asm`?

(Other changes made)

**Comment by luke-jr** [comment]
No, `experimental` is not needed on ARM without `asm`.

ARM users who don't want to rely on not-well-tested assembly code should `USE='-asm'` instead of `USE='experimental'`

**Pull request closed without merging**